### PR TITLE
Added lintDuplicateMetric to promtool

### PR DESCRIPTION
**Pull request by geekodour**
see #5242

### CODE REVIEW

File: util/promlint/promlint.go [hunk 1]
```diff
@@ -101,6 +102,7 @@ func lint(mf dto.MetricFamily) []Problem {
 		lintMetricUnits,
 		lintCounter,
 		lintHistogramSummaryReserved,
+		lintNotUniqueMetric,
```
Comment (tariq1890) [inline util/promlint/promlint.go]:
```suggestion
		lintDuplicateMetric,
```

**Comment by gouthamve** [comment]
Hey, we do want this, but we should check for the timestamps as well and we should make sure that the timestamps are in ascending order (if there are timestamps).

Sorry for waiting so long to get back to this. Do you want to pick this up again?

**Comment by geekodour** [comment]
@gouthamve sure, I'll update it. :) 

Just to confirm this is the format for timestamped metrics right? (Since I've not come across any timestamped metric yet :p)

```
a_total{foo="4"} 1 12345678901234567890.1234567890
```

**Comment by brian-brazil** [comment]
It's a integer number of milliseconds.

**Comment by bwplotka** [comment]
Hello from the Bug Scrub!

Sounds like this needs rebase and @gouthamve comment needs to be addressed https://github.com/prometheus/prometheus/pull/5312#issuecomment-668057564

@bboreham was nice enough to mention he will try to complete this work 💪🏽

**Comment by beorn7** [comment]
Hello from the bug scrub once again.

@bboreham are you still going to work on this?

**Comment by bboreham** [comment]
The code has since moved to a different repo, so I created https://github.com/prometheus/client_golang/pull/1472.

@gouthamve's comment is suggesting a completely different check, so I didn't do that.

I believe there is nothing left to do in this PR; if the new PR is merged, promtool will get updated shortly after.

**Pull request closed without merging**